### PR TITLE
Remove paddle.fluid.layers.utils.*

### PR DESCRIPTION
**Pull request by zhengqiwen1997**
### PR types

Others
### PR changes

APIs
### Describe

Pcard-67005
To clean the code in paddle.fluid, the useless codes should be removed and code still be used should move to a new position.

For function paddle.fluid.layers.utils.convert_to_list, paddle.fluid.layers.utils.is_sequence, paddle.fluid.layers.utils.uniqueidmap, paddle.fluid.layers.utils.uniqueid, paddle.fluid.layers.utils._hash_with_id, paddle.fluid.layers.utils._sorted, paddle.fluid.layers.utils._yield_value, paddle.fluid.layers.utils._yield_flat_nest, paddle.fluid.layers.utils.to_sequence, paddle.fluid.layers.utils.flatten, paddle.fluid.layers.utils._sequence_like, paddle.fluid.layers.utils._packed_nest_with_indices, paddle.fluid.layers.utils.pack_sequence_as, paddle.fluid.layers.utils.map_structure, paddle.fluid.layers.utils.hold_mutable_vars, paddle.fluid.layers.utils.copy_mutable_vars, paddle.fluid.layers.utils._recursive_assert_same_structure, paddle.fluid.layers.utils.padding_to_same_structure, paddle.fluid.layers.utils.assert_same_structure, paddle.fluid.layers.utils._is_symmetric_padding, paddle.fluid.layers._contain_var, paddle.fluid.layers.get_shape_tensor_inputs, paddle.fluid.layers._convert_to_tensor_list, paddle.fluid.layers.convert_shape_to_list, paddle.fluid.layers.check_shape, paddle.fluid.layers.try_set_static_shape_tensor, paddle.fluid.layers.try_get_constant_shape_from_tensor and paddle.fluid.layers.utils.get_inputs_outputs_in_block, since it is not used in version 2.x, should be removed.

For unittest code where this function is used, if it is aimed to test the old API, it can be removed directly. Otherwise, the paddle.fluid.layers.utils.convert_to_list should be replaced with paddle.utils.convert_to_list, paddle.fluid.layers.utils.is_sequence should be replaced with paddle.utils.is_sequence, paddle.fluid.layers.utils.uniqueidmap should be replaced with paddle.utils.layers_utils.uniqueidmap, paddle.fluid.layers.utils.uniqueid should be replaced with paddle.utils.layers_utils.uniqueid, paddle.fluid.layers.utils._hash_with_id should be replaced with paddle.utils._hash_with_id, paddle.fluid.layers.utils._sorted should be replaced with paddle.utils._sorted, paddle.fluid.layers.utils._yield_value should be replaced with paddle.utils._yield_value, paddle.fluid.layers.utils._yield_flat_nest should be replaced with paddle.utils._yield_flat_nest, paddle.fluid.layers.utils.to_sequence should be replaced with paddle.utils.to_sequence, paddle.fluid.layers.utils.flatten should be replaced with paddle.utils.flatten, paddle.fluid.layers.utils._sequence_like should be replaced with paddle.utils._sequence_like, paddle.fluid.layers.utils._packed_nest_with_indices should be replaced with paddle.utils._packed_nest_with_indices, paddle.fluid.layers.utils.pack_sequence_as should be replaced with paddle.utils.pack_sequence_as, paddle.fluid.layers.utils.map_structure should be replaced with paddle.utils.map_structure, paddle.fluid.layers.utils.hold_mutable_vars should be replaced with paddle.utils.hold_mutable_vars, paddle.fluid.layers.utils.copy_mutable_vars should be replaced with paddle.utils.copy_mutable_vars, paddle.fluid.layers.utils._recursive_assert_same_structure should be replaced with paddle.utils._recursive_assert_same_structure, paddle.fluid.layers.utils.padding_to_same_structure should be replaced with paddle.utils.padding_to_same_structure, paddle.fluid.layers.utils.assert_same_structure should be replaced with paddle.utils.assert_same_structure, paddle.fluid.layers.utils._is_symmetric_padding should be replaced with paddle.utils._is_symmetric_padding, paddle.fluid.layers._contain_var should be replaced with paddle.utils._contain_var, paddle.fluid.layers.get_shape_tensor_inputs should be replaced with paddle.utils.get_shape_tensor_inputs, paddle.fluid.layers._convert_to_tensor_list should be replaced with paddle.utils._convert_to_tensor_list, paddle.fluid.layers.convert_shape_to_list should be replaced with paddle.utils.convert_shape_to_list, paddle.fluid.layers.check_shape should be replaced with paddle.utils.check_shape, paddle.fluid.layers.try_set_static_shape_tensor should be replaced with paddle.utils.try_set_static_shape_tensor, paddle.fluid.layers.try_get_constant_shape_from_tensor should be replaced with paddle.utils.try_get_constant_shape_from_tensor and paddle.fluid.layers.utils.get_inputs_outputs_in_block should be replaced with paddle.utils.get_inputs_outputs_in_block.

### CODE REVIEW

File: python/paddle/jit/dy2static/variable_trans_func.py [hunk 1]
```diff
@@ -14,8 +14,8 @@
 
 import paddle
 from paddle.fluid.framework import Variable
-from paddle.fluid.layers.utils import is_sequence, map_structure
 from paddle.utils import gast
+from paddle.utils.layers_utils import is_sequence
```
Comment (zoooo0820) [inline python/paddle/jit/dy2static/variable_trans_func.py]: is_sequence， map_structure是否可以统一一下

**Review by zoooo0820 — DISMISSED** [review]
LGTM

**Comment by jeff41404** [comment]
We will move some internal function `paddle.fluid.layers.utils.*` to file of `python/paddle/utils/layers_utils.py`. I suggest adding some `from .layers_utils import xxx`in python/paddle/utils/\_\_init\_\_.py, so we can use `paddle.utils.*`(not `paddle.utils.layers_utils.*`) instead of `paddle.fluid.layers.utils.*`, it is better?

**Review by zoooo0820 — APPROVED** [review]
LGTM

**Review by wanghuancoder — APPROVED** [review]

**Review by XiaoguangHu01 — APPROVED** [review]
LGTM

**Review by jeff41404 — APPROVED** [review]
LGTM

**Pull request merged**